packages/scene-composer/src/components/panels/scene-components/tag-style/ColorPicker/ColorPicker.tsx

```diff
@@ -0,0 +1,98 @@
+import { Button, FormField, Icon, Input, InputProps, SpaceBetween } from '@awsui/components-react';
+import { NonCancelableCustomEvent } from '@awsui/components-react/internal/events';
+import React, { useCallback, useState } from 'react';
+import { ChromePicker, CirclePicker } from 'react-color';
+
+import { IColorPickerProps } from '../interface';
+
+import {
+  tmAddButton,
+  tmColorPickerContainer,
+  tmCover,
+  tmDivider,
+  tmPopover,
+  tmColorPickerPopover,
+} from './ColorPickerUtils/ColorPickerStyles';
+import { colorPickerPreviewSvg } from './ColorPickerUtils/SvgParserHelper';
+import { colorArray, palleteColors } from './ColorPickerUtils/TagColors';
+
+export const ColorPicker = ({ color, onSelectColor, label }: IColorPickerProps): React.JSX.Element => {
+  const [showPicker, setShowPicker] = useState<boolean>(false);
+  const [showChromePicker, setShowChromePicker] = useState(false);
+  const [selectedColor, setSelectedColor] = useState('');
+  const [_addedColors, setAddedColors] = useState<string[]>([]);
+
+  const handleClick = useCallback(() => {
+    setShowPicker(!showPicker);
+  }, []);
+
+  const handleColorChange = useCallback(
+    (color) => {
+      onSelectColor(color.hex);
+    },
+    [color],
+  );
+
+  const handleHexCodeChange = useCallback(
+    (event: NonCancelableCustomEvent<InputProps.ChangeDetail>) => {
+      onSelectColor(event.detail.value);
+    },
+    [color],
+  );
+
+  const handleShowChromePicker = useCallback(() => {
+    setShowChromePicker(true);
+    setShowPicker(false);
+  }, []);
+
+  const handleClose = useCallback(() => {
+    setShowPicker(false);
+  }, []);
+
+  const handleCloseChromePicker = useCallback(() => {
+    setShowChromePicker(false);
+  }, []);
+
+  return (
+    <SpaceBetween size='m'>
+      <SpaceBetween size='m' direction='horizontal'>
+        <FormField label={label}></FormField>
+        <Button
+          data-testid='color-preview'
+          ariaLabel='color-picker-preview'
+          variant='inline-icon'
+          iconSvg={<Icon size='big' svg={colorPickerPreviewSvg(color)} />}
+          onClick={() => {
+            handleClick();
+            onSelectColor(color);
+          }}
+        />
+        <Input ariaLabel='hexcode' data-testid='hexcode' value={color} onChange={handleHexCodeChange} />
+        <div style={tmColorPickerContainer}>
+          {showPicker && !showChromePicker && (
+            <div style={tmColorPickerPopover} onClick={handleClose}>
+              <CirclePicker
+                width='300px'
+                data-testid='circlePicker'
+                aria-label='color'
+                colors={colorArray(palleteColors)}
+                color={color}
+                onChange={handleColorChange}
+              />
+              <div style={tmDivider} />
+              <button style={tmAddButton} onClick={handleShowChromePicker}>
+                +
+              </button>
```

Comment:
can we use the add button from awsui?

Reply:
or at least an awsui icon? otherwise the button will look different when consumers have different font settings in their browsers

Reply:
done